### ISSUE TITLE
scheduler filter include streaming part

### ISSUE THREAD

**Issue by szuecs** [zalando/skipper#2579]
**Is your feature request related to a problem? Please describe.**

As owner of a streaming application that only support N concurrent users I want to be able to limit the traffic to N.
Right now if you do have a backend that does async response where the streaming part can be long, you get much more than N requests, because the scheduler filters response part is triggered before the response body streaming is happening.  

**Describe the solution you would like**

Make it possible to limit streaming concurrency, too.
Basically decrease the concurrency counter after body streaming was done.

**Describe alternatives you've considered (optional)**

We tried `lifo` which does not work as expected and a workaround seems to be `backendRatelimit` which has a lower error margin than lifo for an application which streams the response.

**Additional context (optional)**
internal finding

**Comment by AshrafAhmed9** [comment]
Looks like this got half done. #2685 added `fifoWithBody`, which releases the slot from the proxy after the body is streamed instead of in the filter. `lifo` never got that — `lifoFilter.Response()` still calls `response(scheduler.LIFOKey, ctx)` straight away, so the reservation is gone before anything is written. Which is the case in the original report, since it mentions lifo not holding the limit.

Want me to add a `lifoWithBody` mirroring the fifo one?

One thing I'd want your call on first. The proxy hook is keyed on the filter name:

```go
// fifoWithBody() filter
if sbf, ok := ctx.StateBag()[filters.FifoWithBodyName]; ok {
```

So either I add a second lookup next to it for lifo, or turn it into one list that any scheduler filter registers into. The second is cleaner but it touches the existing fifo path, so I'd default to the first unless you'd rather not have that block copied a third time later.

Also, `lifoGroup` as well or just plain `lifo` to start?

For tests I'd hold the response body open and assert inflight stays at the configured concurrency while it streams — that should fail on lifo as it is today, so the test shows the gap before it fixes it.

**Comment by szuecs** [comment]
Yeah if you implement, do both lifo and lifoGroup WithBody alongside the fifo.
